feat: shield settings fixes

lionellbriones opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Fixes for metamask shield settings

For annual subscription change text from `Yearly` to `Annual`
https://consensyssoftware.atlassian.net/browse/SUBS-476

Don show skeleton loader before redirecting to `shield plan` from `Settings > Transaction shield`
https://consensyssoftware.atlassian.net/browse/SUBS-471

Remove `Resubscribe` button on banner and add `Restart membership` on action list
https://consensyssoftware.atlassian.net/browse/SUBS-474
https://consensyssoftware.atlassian.net/browse/SUBS-479

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/36284?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: Fixes for Metamask Transaction Shield

## **Related issues**

Fixes:

## **Manual testing steps**

1. Testing steps available on jira task links provided

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**
Go to shield settings without subscription (on current state) - redirect to shield-plan
https://github.com/user-attachments/assets/cf831d8a-0aac-4448-b267-3cd23b7aff1d

Subscribe from shield-plan and redirect to settings
https://github.com/user-attachments/assets/7892f302-a044-48e0-8c14-4d60c207fcb6

Has shield subscription and go to settings
https://github.com/user-attachments/assets/5437442a-588e-414e-a222-6c8723ec801f

Show generic loader when doing cancel, restart membership, view billing history
https://github.com/user-attachments/assets/44e33f1d-36a2-46cb-8f0c-aec37470552c

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

lionellbriones committed:
feat: added shield plan UI

lionellbriones committed:
feat: updated font size

lionellbriones committed:
feat: add save badge, custom token list and confirmation style UI

lionellbriones committed:
feat: added locales

lionellbriones committed:
feat: fix states

lionellbriones committed:
feat: added shield payment modal tests

lionellbriones committed:
feat: add shield plan test

lionellbriones committed:
Merge branch 'main' into feat/shield-plan

lionellbriones committed:
feat: filter stable coins
- change contents based on payment method
- show usdt usdc and musd only with balance

lionellbriones committed:
Merge branch 'main' into feat/shield-plan

lionellbriones committed:
Merge branch 'main' into feat/shield-plan

lionellbriones committed:
feat: fix lint and unit tests

lionellbriones committed:
feat: fix lint and null selected asset

lionellbriones committed:
fix: test lint

lionellbriones committed:
fix: removed unused codes

lionellbriones committed:
fix: issue on token list generator

lionellbriones committed:
feat: update code owner

lionellbriones committed:
feat: remove code owners
- add todo on shield plan and type

tuna1207 committed:
feat: subscription controller init

lionellbriones committed:
Merge branch 'main' into feat/shield-plan

lionellbriones committed:
feat: fix messages lint

tuna1207 committed:
feat: expose subscription api

lionellbriones committed:
feat: add loader

tuna1207 committed:
feat: handle shield payment plan tokens

tuna1207 committed:
fix: correct pricing hook

tuna1207 committed:
feat: update actions

tuna1207 committed:
feat: shield plan subscription data

tuna1207 committed:
feat: handle controller method

tuna1207 committed:
feat: update shield screen integration

tuna1207 committed:
feat: update billing display

tuna1207 committed:
feat: update user subscriptions hook

lionellbriones committed:
Merge branch 'main' into feat/shield-plan

tuna1207 committed:
Merge branch 'feat/shield-plan' into feat/subscription-payment-options-integration

tuna1207 committed:
Merge branch 'feat/subscription-payment-options-integration' into feat/subscription-shield-settings-integration

tuna1207 committed:
feat: move launchwebauthflow to background

tuna1207 committed:
fix: correct trial requested

tuna1207 committed:
fix: lowercase address compare

tuna1207 committed:
fix: typo

tuna1207 committed:
fix: correct comment

tuna1207 committed:
fix: use selected token icon

tuna1207 committed:
Merge branch 'feat/subscription-payment-options-integration' into feat/subscription-shield-settings-integration

tuna1207 committed:
Merge branch 'main' into feat/subscription-payment-options-integration

tuna1207 committed:
chore: temp controller tgz

tuna1207 committed:
fix: lint

tuna1207 committed:
fix: test case

tuna1207 committed:
fix: correct log import

tuna1207 committed:
fix: shield plan story

tuna1207 committed:
fix: undefined subscriptions

tuna1207 committed:
chore: use correct subscription controller version

tuna1207 committed:
fix: lint locale

tuna1207 committed:
fix: shield plan text

tuna1207 committed:
chore: yarn dedupe

metamaskbot committed:
Update LavaMoat policies

tuna1207 committed:
fix: start subscription with card params

tuna1207 committed:
chore: update lavamoat

tuna1207 committed:
Merge branch 'feat/subscription-payment-options-integration' into feat/subscription-shield-settings-integration

metamaskbot committed:
Update LavaMoat policies

tuna1207 committed:
Merge branch 'feat/subscription-payment-options-integration' into feat/subscription-shield-settings-integration

tuna1207 committed:
feat: update e2e state check

tuna1207 committed:
fix: lint

tuna1207 committed:
Merge branch 'feat/subscription-payment-options-integration' into feat/subscription-shield-settings-integration

lionellbriones committed:
feat: stop showing loader on view billing
- hide cancel modal after clicking cancel
- hide unsubscribe when loading

tuna1207 committed:
Merge branch 'main' into feat/subscription-shield-settings-integration

tuna1207 committed:
feat: use customer id for member ship id

tuna1207 committed:
feat: cancel subscription button show and date

lionellbriones committed:
feat: updated locales banner alert and payment method button

lionellbriones committed:
feat: added paused state

lionellbriones committed:
feat: update locales and copywriting
- added priority tag on support

tuna1207 committed:
chore: subscription controller dev build use dev backend

tuna1207 committed:
fix: correct payment method card

lionellbriones committed:
feat: add a loader when user submit shield plan

tuna1207 committed:
refactor: start subscription pop up loading

tuna1207 committed:
fix: lint

tuna1207 committed:
fix: correct sheild plan when continue

tuna1207 committed:
fix: correct platform usage

tuna1207 committed:
fix: test cases

tuna1207 committed:
fix: trialed param

tuna1207 committed:
fix: trailed check test

tuna1207 committed:
fix: correct start subscription function

tuna1207 committed:
chore: add todo comment

lionellbriones committed:
feat: change yearly to annual

lionellbriones committed:
feat: remove resubscribe and add restart membership

lionellbriones committed:
feat: redirect to shield plan if no subscription yet before loading page

lionellbriones committed:
feat: fixed getSubscription prop

lionellbriones committed:
Merge branch 'main' into feat/shield-settings-fixes

lionellbriones committed:
Merge branch 'main' into feat/shield-settings-fixes

lionellbriones committed:
feat: remove yearly from locales

lionellbriones committed:
Merge branch 'main' into feat/shield-settings-fixes

lionellbriones committed:
feat: remove changes which is fix on another PR

chaitanyapotti commented on ui/pages/settings/settings.container.js:
```diff
@@ -144,12 +145,19 @@ const mapStateToProps = (state, ownProps) => {
+  const { subscriptions } = getUserSubscriptions(state);
+  // get shield subscription
+  const shieldSubscription = subscriptions.find((subscription) =>
+    subscription.products.some((product) => product.name === 'shield'),
```
isn't the product name a constant? if not, we should make it one and it should be in core package
@tuna1207

chaitanyapotti replied:
can do in the next pr along with core update

lionellbriones committed:
Merge branch 'main' into feat/shield-settings-fixes

lionellbriones committed:
Merge branch 'main' into feat/shield-settings-fixes

lionellbriones committed:
feat: remove checking subscription on settings container

lionellbriones committed:
feat: redirect shield-plan if no subscription on current state
- show skeleton loader on page load
- show generic loader on cancel billing history and resubscribe

lionellbriones commented on ui/pages/settings/transaction-shield-tab/transaction-shield.tsx:
```diff
@@ -505,14 +493,14 @@ const TransactionShield = () => {
                         maximumFractionDigits: 0,
                       },
-                    )} (${shieldSubscription.interval === RECURRING_INTERVALS.year ? t('shieldPlanYearly') : t('shieldPlanMonthly')})`,
+                    )} (${shieldSubscription.interval === RECURRING_INTERVALS.year ? t('shieldPlanAnnual') : t('shieldPlanMonthly')})`,
```
it was added before this PR

lionellbriones commented on ui/pages/settings/transaction-shield-tab/transaction-shield.tsx:
```diff
@@ -107,17 +107,18 @@ const TransactionShield = () => {
       navigate(SHIELD_PLAN_ROUTE);
     }
-  }, [navigate, loading, shieldSubscription]);
+  }, [navigate, shieldSubscription]);
```
Yes it will not wait for subscriptions to be updated and just base the behavior on current subscription state. After redirecting to shield-plan page it will fetch updated subscription state (if has subscription it will return back to settings, if not it will stay to shield-plan page)

chaitanyapotti replied:
Will be fixed in upcoming pr
